media-source-respec.html

```diff
@@ -2719,6 +2711,280 @@ <h3>Event Summary</h3>
+      <ol>
+        <li>
+          <dl class="switch">
+            <dt>
+              For each |buffer:ManagedSourceBuffer| in [=this=]'s
+              {{MediaSource/sourceBuffers}}:
+            </dt>
+            <dd>
+              <ol>
+                <li>Run the |buffer|'s [=ManagedSourceBuffer/memory cleanup=] algorithm.</li>
+              </ol>
+            </dd>
+          </dl>
+        </li>
+      </ol>
+    </section>
+
+    <section data-dfn-for="BufferedChangeEvent">
+      <h2><dfn>BufferedChangeEvent</dfn> interface</h2>
+      <pre class="idl">
+        [Exposed=(Window,DedicatedWorker)]
+        interface BufferedChangeEvent : Event {
+          constructor(DOMString type, BufferedChangeEventInit eventInitDict);
+
+          [SameObject] readonly attribute TimeRanges addedRanges;
+          [SameObject] readonly attribute TimeRanges removedRanges;
+        };
+
+        dictionary BufferedChangeEventInit : EventInit {
+          TimeRanges addedRanges;
+          TimeRanges removedRanges;
+        };
+      </pre>
+      <h3>Attributes</h3>
+      <dl>
+        <dt>
+          <dfn>addedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges added during the last run of the [=coded frame processing=] algorithm.
+        </dd>
+        <dt>
+          <dfn>removedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges removed during the last run of the [=coded frame removal=] or [=coded frame eviction=] algorithm or if the
+          user agent evicted content in response to a [=ManagedSourceBuffer/memory cleanup=].
+        </dd>
+      </dl>
+    </section>
+
+    <section>
+      <h2><dfn>ManagedSourceBuffer</dfn> interface</h2>
+      <pre class="idl">
+        [Exposed=(Window,DedicatedWorker)]
+        interface ManagedSourceBuffer : SourceBuffer {
+          attribute EventHandler onbufferedchange;
+        };
+      </pre>
+      <h3>Attributes</h3>
+      <dl class="attributes" data-dfn-for="ManagedSourceBuffer">
+        <dt>
+          <dfn>onbufferedchange</dfn>
+        </dt>
+        <dd>
+          <p>An [=event handler IDL attribute=] whose [=event handler event type=] is {{bufferedchange}}.</p>
+        </dd>
+      </dl>
+
+      <h3>Event Summary</h3>
+      <table class="old-table">
+        <thead>
+          <tr>
+            <th>Event name</th>
+            <th>Interface</th>
+            <th>Dispatched when...</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <dfn class="event">bufferedchange</dfn>
+            </td>
+            <td>{{Event}}</td>
+            <td>
+              The {{ManagedSourceBuffer}}'s buffered range changed following a
+              call to {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}},
+              or as a consequence of the user agent running the
+              [=ManagedSourceBuffer/memory cleanup=] algorithm.
```

Comment:
Here, in the BufferedChangeEvent's addedRanges and removedRanges, and in the "Buffered Change" algorithm, below, effective change of ManagedSourceBuffer's buffered attribute can come from other operations not mentioned.
Beyond appendBuffer, remove and memory cleanup, a ManagedSourceBuffer's buffered range additions and removals can be caused by open->ended MediaSource readyState transition (on endOfStream(<no error>)) and from ended->open transition, respectively: intersection ranges logic in the calculation of the SourceBuffer buffered attribute is conditioned on readyState: if ended, then "jagged ended" muxed tracks' highest end time is used for all those tracks.